presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -712,6 +712,15 @@ public PlanWithProperties visitJoin(JoinNode node, PreferredProperties preferred
                     .collect(toImmutableList());
 
             JoinNode.DistributionType distributionType = node.getDistributionType().orElseThrow(() -> new IllegalArgumentException("distributionType not yet set"));
+
+            // use partitioned join if probe side is naturally partitioned on join symbols (e.g: because of aggregation)
+            if (!node.getCriteria().isEmpty()) {
+                PlanWithProperties left = node.getLeft().accept(this, PreferredProperties.partitioned(ImmutableSet.copyOf(leftSymbols)));
```

Comment:
Should we try here:
```
node.getLeft().accept(this, PreferredProperties.partitioned(ImmutableSet.copyOf(leftSymbols)));
```
or (broadcast approach)
```
node.getLeft().accept(this, PreferredProperties.any())
```

Even though `PreferredProperties.partitioned(...)` doesn't enforce any partitioning it might cause different planning (e.g: `AddExchanges.Rewriter#visitUnion`).

CC: @martint ?

Reply:
Ok so I changed it with:
```
node.getLeft().accept(this, PreferredProperties.any())
```
This is safest thing to do